plugins/invite.lua

```diff
@@ -7,41 +7,45 @@ do
 local function run(msg, matches)
   -- User submitted a user name
   if matches[1] == "name" then
-    local user = matches[2]
-    user = string.gsub(user," ","_")
-  end
-  
+    user_ = matches[2]
+    user_ = string.gsub(user_," ","_")  
   -- User submitted an id
-  if matches[1] == "id" then
-    local user = matches[2]
-    user = 'user#id'..user
+  elseif matches[1] == "id" then
+    user_ = matches[2]
+    user_ = 'user#id'..user_
   end
 
   -- The message must come from a chat group
   if msg.to.type == 'chat' then
-    local chat = 'chat#id'..msg.to.id
+    chat_id_ = 'chat#id'..msg.to.id
   else 
     return 'This isnt a chat group!'
   end
 
-  print ("Trying to add: "..user.." to "..chat)
-  local status = chat_add_user (chat, user, ok_cb, false)
-  if not status then
+  print ("Trying to add: "..user_.." to "..chat_id_)
+  local success = chat_add_user (chat_id_, user_, ok_cb, false)
+  if not success then
+    user_ = nil
+    chat_id_ = nil
     return "An error happened"
+  else
+    local added = "Added user: "..user_.." to "..chat_id_
+    user_ = nil
+    chat_id_ = nil
+    return added
   end
-  return "Added user: "..user.." to "..chat
 end
 
 return {
-  description = "Invite other user to the chat group", 
+  description = "Invite other user to the chat group",
   usage = {
     "!invite name [user_name]", 
     "!invite id [user_id]" },
   patterns = {
-    "^!invite (name) (.*)$",
-    "^!invite (id) (%d+)$"
+    "^!invite (name) (.*)",
+    "^!invite (id) (%d+)"
```

Comment:
No, the $ at the end is important, if not the pattern match can get lines that shouldn't
For example, the second one without the dollar match this command, and shouldn't:
`!invite id 123123123 hi :)`